### PR TITLE
Feat/#4585

### LINKED ISSUE #4585: Create a script to set up the project
This task is a preparation for building CLI tool that will generate working Vue Storefront project.
------
Currently the only way to use Vue Storefront Next with working boilerplate is by copying the whole repository (together with all integrations) which is (obviously) not suitable for production usage.

Our goal is to let our users generate fresh VSF Next projects through CLI. Before that we should prepare a script that will be used by the CLI. Fresh project should have all files exposed to the user (in opposite to core theme inheritance)

# Preparation steps
1. Rename all themes to follow `@vue-storefront/{integration}-theme` pattern. (now all are named just `theme`
2. create a new package in `core` folder called `cli`, name it `@vue-storefront/cli` with version `2.0`
3. Inside the package create a folder `src/scripts/createProject.ts` which will be a working directory for this specific task

# What steps are required to generate the project?

A baseline for the new project will be `theme` folder from the integration along with  agnostic files in `theme-module`. To generate a fresh project a script should perform following steps (1 i 2 can be done in reversed order if you find it easier to achieve):

**Important**: Remember that the script will run from the CLI. Use NPM package paths for integration themes instead of repo paths (eg `@vue-storefront/commercetools-theme`
1. **Copy content of `{integration}/theme folder`** into a new project which will be it's base containing all configuration files and some theme-specific ones. ({integration} should be a parameter so we can easily switch between copying `about-you-theme` and )
2. **Copy and compile content from agnostic theme** (`theme-module/theme`). In case of conflicting files don't copy them (or you will override theme files)!
3. **Remove `theme-module` from `nuxt-config.js`** to prevent generation of `.theme` folder (but keep it in `package.json` to inherit core dependencies in new package
```    
['@vue-storefront/nuxt-theme', {
      apiClient: '@vue-storefront/commercetools-api',
      composables: '@vue-storefront/commercetools'
    }]
```
4. remove `coreDevelopment: true,` in `nuxt-config.js`

### PR DESCRIPTION
**Pull request by Fifciu**
Close #4585

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -41,7 +41,8 @@
     "dev:bp": "cd packages/boilerplate/theme && yarn dev",
     "dev:ct": "cd packages/commercetools/theme && yarn dev",
     "link-packages": "lerna link --force-local",
-    "lint": "eslint . --ext .js,.ts,.vue"
+    "lint": "eslint . --ext .js,.ts,.vue",
+    "test-cli": "cd packages/core/cli/src/scripts && ts-node createProject.ts"
```
Comment (filrak) [inline package.json:45]: **change:** Imho it's not needed here. If you want to test it - write a test ;)

File: packages/about-you/theme/nuxt.config.js [hunk 2]
```diff
@@ -43,10 +43,12 @@ export default {
         ]
       }
     }],
+    // @core-development-only-start
```
Comment (filrak) [inline packages/about-you/theme/nuxt.config.js:46]: **missing docs:** Please include a paragraph about this magic comment in theme docs

File: packages/core/cli/package.json [hunk 3]
```diff
@@ -0,0 +1,17 @@
+{
+  "name": "@vue-storefront/cli",
```
Comment (filrak) [inline packages/core/cli/package.json:2]: **change:** please mark this package as private for now

File: packages/core/cli/src/scripts/createProject.ts [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    return Promise.all(getAllFilesFromDir(filesDir as Array<string>).map(
+      file => copyFile(file, buildFileTargetPath(file, targetPath, chopPhrase))
+    ));
+  } else {
+    return copyFile(filesDir, buildFileTargetPath(filesDir as string, targetPath, chopPhrase));
+  }
+};
+
+const removeDevMagicComment = (source: string): string => source.replace(/\s+(\/\/ @core-development-only-start)(.*?)(\/\/ @core-development-only-end)/sg, '');
+const removeCoreDevelopmentMode = (source: string): string => source.replace(/coreDevelopment:(\s)*?true[,]?/, '');
```
Comment (filrak) [inline packages/core/cli/src/scripts/createProject.ts:30]: *8suggestion** can be done with magic comment as well. By doing this you're keeping same convention for all things that should be removed in clients project

File: package.json (hunk 1 above)
Comment (andrzejewsky) [reply package.json:45]: that's not needed,  please use `bin` section in the `package.json` of CLI package, then once you have installed package you can just use `yarn [your cli name]` () and that's it!

```
"bin": {
  "vsf": "ts-nodd cli.ts"
}
```

Comment (andrzejewsky) [reply package.json:45]: btw. with using this we probably will be able to use this package with npx

Comment (filrak) [reply package.json:45]: @andrzejewsky yup thats the next step ;)

Comment (filrak) [reply package.json:45]: also you can call @Fifciu "Jędrek" or "Jendre" (more international).

Comment (Fifciu) [reply package.json:45]: Call me "Fifciuu" guys. Double-U is important. Nonetheless, Jendre sounds funny :D

File: packages/core/docs/contributing/themes.md [hunk 5]
```diff
@@ -10,18 +10,28 @@ Default theme is located in `packages/core/theme-module` folder and recognized a
 
 ## Configuration
 
-To inherit from the default theme in your integration theme you need to install private `@vue-storefront/nuxt-theme` package. from this repo.
+Run VSF CLI create project command with 2 arguments. The first is name of the integration, e.g. `commercetools`, `about-you` or just `boilerplate`. Latter is the target path where your project will be created.
```
Comment (filrak) [inline packages/core/docs/contributing/themes.md:13]: There is no such command yet. Also this document is meant to explana how themes are working not how to create a project and... You deleted that part :d

File: packages/core/cli/src/scripts/createProject.ts [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+};
+
+const buildFileTargetPath = (file: string, targetPath: string, chopPhrase: string): string => targetPath + (file.replace(chopPhrase, ''));
+
+const copyThemeFiles = (filesDir: string | Array<string>, targetPath: string, chopPhrase: string) => {
+  if (fs.statSync(filesDir).isDirectory()) {
+    return Promise.all(getAllFilesFromDir(filesDir as Array<string>).map(
+      file => copyFile(file, buildFileTargetPath(file, targetPath, chopPhrase))
+    ));
+  } else {
```
Comment (andrzejewsky) [inline packages/core/cli/src/scripts/createProject.ts:24]: you don't need that else here

File: packages/core/cli/src/scripts/createProject.ts [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+    .map(directory => path.join(integrationThemePath, directory));
+
+  log.info(`Coppying ${integration}-theme to ${targetPath}`);
+  await Promise.all(
+    integrationThemeFiles.map(
+      absoluteDirectoryPath => copyThemeFiles(absoluteDirectoryPath, absoluteTargetPath, integrationThemePath)
+    )
+  );
+
+  const agnosticThemePath = '../../node_modules/@vue-storefront/nuxt-theme/theme';
```
Comment (andrzejewsky) [inline packages/core/cli/src/scripts/createProject.ts:46]: it's very similar to the line 32, so create just a function called eg. 
```
const getThemePath = (themeName) => `../../node_modules/@vue-storefront/${themeName}`;
```

and use:
```
const integrationThemePath = getThemePath(`${integration}-theme`);
const agnosticThemePath = getThemePath('nuxt-theme/theme');
```

File: packages/core/cli/src/scripts/createProject.ts [hunk 8]
```diff
@@ -0,0 +1,76 @@
+const compileTemplate = require('@vue-storefront/nuxt-theme/scripts/compileTemplate');
+const getAllFilesFromDir = require('@vue-storefront/nuxt-theme/scripts/getAllFilesFromDir.js');
+const { copyFile } = require('@vue-storefront/nuxt-theme/scripts/copyThemeFiles');
+const consola = require('consola');
+const chalk = require('chalk');
+
+const fs = require('fs');
+const path = require('path');
+
+const log = {
```
Comment (andrzejewsky) [inline packages/core/cli/src/scripts/createProject.ts:10]: move it to the separated file, that will be useful across whole cli

File: packages/core/cli/src/scripts/createProject.ts [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+  await Promise.all(
+    integrationThemeFiles.map(
+      absoluteDirectoryPath => copyThemeFiles(absoluteDirectoryPath, absoluteTargetPath, integrationThemePath)
+    )
+  );
+
+  const agnosticThemePath = '../../node_modules/@vue-storefront/nuxt-theme/theme';
+  const agnosticThemeFiles = getAllFilesFromDir(agnosticThemePath).filter(file => !file.includes(path.sep + 'static' + path.sep));
+
+  const compileAgnosticTemplate = (filePath: string, targetPath: string, chopPhrase: string) => {
```
Comment (andrzejewsky) [inline packages/core/cli/src/scripts/createProject.ts:49]: what's the return type here?

**Pull request merged**